### PR TITLE
feat: Add Message Summarizer Feature

### PR DESCRIPTION
**Pull request by Jack-mi**
# Add Message Summarizer Feature

## Description
Adds a new message summarization feature to generate structured summaries of chat conversations, including both implementation and tests.

## Changes
- Added `message_summarizer.py` with:
  - `SummarySchema`: Structured output for roles, entities, decisions, progress, and context
  - `MessageSummarizer`: Handles message summarization with model support and JSON validation
- Added comprehensive tests in `test_message_summarizer.py` covering:
  - Initialization with different models
  - Basic message summarization
  - Edge cases (empty/single messages)
  - Special character handling

## Testing
```bash
pytest test/utils/test_message_summarizer.py -v -s
```

## Example
```python
summarizer = MessageSummarizer()
summary = summarizer.summarize(messages)
print(f"Roles: {summary.roles}")
print(f"Key Entities: {summary.key_entities}")
```

### CODE REVIEW

**Comment by Jack-mi** [comment]
@jajajalalalala @Michaelsqj Hey guys, Would you be so kind as to assist in reviewing this PR? I have successfully passed all the tests.

File: camel/utils/message_summarizer.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+from camel.agents import ChatAgent
+from camel.messages import BaseMessage
+from camel.models import ModelFactory
+from camel.types import ModelPlatformType, ModelType
+from camel.utils.response_format import model_from_json_schema
+
+
+class SummarySchema(BaseModel):
+    """Schema for structured message summaries.
```
Comment (Wendong-Fan) [inline camel/utils/message_summarizer.py]:
```suggestion
    r"""Schema for structured message summaries.
```

File: camel/utils/message_summarizer.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    Attributes:
+        roles (List[str]): The roles involved in the conversation
+        key_entities (List[str]): Important entities/concepts discussed
+        decisions (List[str]): Key decisions or conclusions reached
+        task_progress (str): Summary of progress made on the main task
+        context (str): Additional relevant contextual information
+    """
```
Comment (Wendong-Fan) [inline camel/utils/message_summarizer.py]:
```suggestion
    Attributes:
        roles (List[str]): The roles involved in the conversation.
        key_entities (List[str]): Important entities/concepts discussed.
        decisions (List[str]): Key decisions or conclusions reached.
        task_progress (str): Summary of progress made on the main task.
        context (str): Additional relevant contextual information.
    """
```

File: camel/utils/message_summarizer.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+    roles: List[str]
+    key_entities: List[str]
+    decisions: List[str]
+    task_progress: str
+    context: str
+
+
+class MessageSummarizer:
+    """Utility class for generating structured summaries of chat messages.
```
Comment (Wendong-Fan) [inline camel/utils/message_summarizer.py]:
```suggestion
    r"""Utility class for generating structured summaries of chat messages.
```

File: camel/utils/message_summarizer.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+    Args:
+        model (Union[str, ModelType], optional):
+            The model to use for summarization.
+            Defaults to ModelType.DEFAULT.
```
Comment (Wendong-Fan) [inline camel/utils/message_summarizer.py]: would be better to make `BaseModelBackend` as argument instead of hardcode the model platform as default?

File: camel/utils/message_summarizer.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+                "- Progress made on the main task\n"
+                "- Relevant contextual information\n\n"
+                "Provide summaries that are concise while "
+                "preserving critical information.",
+            ),
+            model=self.model_backend,
+        )
+
+    def summarize(self, messages: List[BaseMessage]) -> SummarySchema:
+        """Generate a structured summary of the provided messages.
```
Comment (Wendong-Fan) [inline camel/utils/message_summarizer.py]:
```suggestion
        r"""Generate a structured summary of the provided messages.
```

File: camel/utils/message_summarizer.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+        """
+        # When messages is empty
+        if len(messages) == 0:
+            return SummarySchema(
+                roles=[],
+                key_entities=[],
+                decisions=[],
+                task_progress="",
+                context="",
+            )
```
Comment (Wendong-Fan) [inline camel/utils/message_summarizer.py]: would raise error be better?

File: camel/utils/message_summarizer.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+                    "description": "Additional relevant \
+contextual information",
+                },
+            },
+        }
+
+        # Create a Pydantic model from the schema
+        ResponseModel = model_from_json_schema(
+            "ResponseModel", response_schema
+        )
```
Comment (Wendong-Fan) [inline camel/utils/message_summarizer.py]: would define the Pydantic model directly would be more straightforward?

File: camel/utils/message_summarizer.py [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+            "- context: additional relevant context"
+        )
+
+        # Get structured summary from model with forced JSON response
+        response = self.agent.step(prompt, response_format=ResponseModel)
+        try:
+            # The response should already be validated by the model
+            content = response.msg.content
+            if isinstance(content, str):
+                content = json.loads(content)
```
Comment (Wendong-Fan) [inline camel/utils/message_summarizer.py]: use `response.msg.parsed` directly?

**Review by Wendong-Fan — COMMENTED** [review]
thanks @Jack-mi 's contribution! Left some comments below

File: camel/utils/message_summarizer.py (hunk 8 above)
Comment (Jack-mi) [reply camel/utils/message_summarizer.py]: 👌updated

File: camel/utils/message_summarizer.py (hunk 7 above)
Comment (Jack-mi) [reply camel/utils/message_summarizer.py]: 👌updated

File: camel/utils/message_summarizer.py (hunk 6 above)
Comment (Jack-mi) [reply camel/utils/message_summarizer.py]: 👌updated

File: camel/utils/message_summarizer.py (hunk 4 above)
Comment (Jack-mi) [reply camel/utils/message_summarizer.py]: 👌updated
